### PR TITLE
[Analytics] Make analytics pipelines run from any checkpoint to support backfill

### PR DESCRIPTION
**Pull request by sadhansood**
## Description 

This PR allows for us to run analytics pipeline from any given checkpoint passed in through the config. This is useful for testing purposes and for backfill we need to be able to run from an arbitrary checkpoint.

## Test Plan 

Ran locally, pipeline works as expected

### CODE REVIEW

File: crates/sui-analytics-indexer/src/lib.rs [hunk 1]
```diff
@@ -459,13 +478,17 @@ pub async fn make_move_call_processor(
...
-    let (epoch, starting_checkpoint_seq_num) =
-        read_store_for_epoch_and_checkpoint(config.remote_store_config.clone(), FileType::MoveCall)
-            .await?;
+    let starting_checkpoint_seq_num = if let Some(starting_checkpoint_seq_num) =
+        config.starting_checkpoint_seq_num
+    {
+        starting_checkpoint_seq_num
+    } else {
+        read_store_for_checkpoint(config.remote_store_config.clone(), FileType::Checkpoint).await?
+    };
```
Comment (dariorussi) [inline crates/sui-analytics-indexer/src/lib.rs:487]: wonder if a function for this wouldn't be more clear

**Review by dariorussi — APPROVED** [review]
minor comments. Let me know if there are reasons not to do that and fine doing it another time if we want it (or now as you wish)

File: crates/sui-analytics-indexer/src/lib.rs [hunk 2]
```diff
@@ -459,13 +478,17 @@ pub async fn make_move_call_processor(
...
     let writer: Box<dyn AnalyticsWriter<MoveCallEntry>> = match config.file_format {
         FileFormat::CSV => Box::new(CSVWriter::new(
             &config.checkpoint_dir,
-            epoch,
+            0,
```
Comment (phoenix-o) [inline crates/sui-analytics-indexer/src/lib.rs:491]: nit: FWIU we rely on the following reset call to set the epoch properly. Is there any reason to keep this parameter in the constructor? (or should we change internal `epoch` field to `Option<u64>`)

Comment (sadhansood) [reply crates/sui-analytics-indexer/src/lib.rs:491]: yeah good call, we can just make epoch initialize with 0 without making it a constructor argument

File: crates/sui-analytics-indexer/src/lib.rs (hunk 1 above)
Comment (sadhansood) [reply crates/sui-analytics-indexer/src/lib.rs:487]: makes sense

**Pull request merged**